### PR TITLE
Add replication instructions for training

### PR DESCRIPTION
**Pull request by haileyschoelkopf**
This PR creates a `pretraining/` folder which contains files and a README for use in replicating our training setup.

### CODE REVIEW

File: pretraining/README.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+```
+
+
+### Check Out Codebase
+
+Next, check out the commit used to train the model you are replicating.
+
+* 7b: https://github.com/EleutherAI/gpt-neox/tree/math-lm-2-rotary
+
+* 34b: https://github.com/EleutherAI/gpt-neox/commit/e59c873ee779df2d7f182deb6ad34f290a077ea4
```
Comment (zhangir-azerbayev) [inline pretraining/README.md:46]: Why do we need to link to two different commits? Isn't it the case that math-lm-2-rotary behaves the same as e59c87 when "attention_type": "multihead"?

File: pretraining/README.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+### Launching Training
+
+Then, edit the provided YML files to set paths based on your own system's saved locations for checkpoints and data files, and edit the SLURM job script as specified (using ) or run the job across multiple nodes using your own system's orchestration.
```
Comment (zhangir-azerbayev) [inline pretraining/README.md:50]: Would adding 
> Note that the global batch size will be scaled by your number of nodes. Therefore, if running on a number of nodes different from 32 you should scale gradient accumulation steps accordingly

be helpful, or is it obvious enough?

**Review by zhangir-azerbayev — COMMENTED** [review]
Looks great. Left two minor comments.

File: pretraining/README.md (hunk 1 above)
Comment (zhangir-azerbayev) [reply pretraining/README.md:46]: Or instead of the `pretraining` directory, should we update the neox submodule to e59c87 and include the instructions in that readme?

File: pretraining/README.md (hunk 2 above)
Comment (haileyschoelkopf) [reply pretraining/README.md:50]: I was considering whether to add this. I think it's worth restating.

File: pretraining/README.md (hunk 1 above)
Comment (haileyschoelkopf) [reply pretraining/README.md:46]: I think having this directory is easier to locate than writing it in the submodule's readme. I left it as separate commits because I figured we should link to the exact commits used, but you're right that behavior is the same. Will update the submodule to e59c87 and unify these

File: pretraining/README.md (hunk 2 above)
Comment (haileyschoelkopf) [reply pretraining/README.md:48]: added

File: pretraining/README.md (hunk 1 above)
Comment (haileyschoelkopf) [reply pretraining/README.md]: done!

**Comment by haileyschoelkopf** [comment]
Resolved comments!
